packages/main/src/MenuItem.ts

```diff
@@ -253,24 +267,90 @@ class MenuItem extends ListItem implements IMenuItem {
 		return false;
 	}
 
+	get isGroup(): boolean {
+		return false;
+	}
+
+	get _parentGroup() {
+		const parent = this.parentElement;
+		return parent && "isGroup" in parent && parent.isGroup ? parent as MenuItemGroup : null;
+	}
+
+	get _parentItemSelectionMode() {
+		return this._parentGroup ? this._parentGroup?.itemSelectionMode : ItemSelectionMode.None;
+	}
+
+	get _list() {
+		return this.shadowRoot!.querySelector<List>("[ui5-list]")!;
+	}
+
+	get _menuItemsNavigation() {
+		const items: MenuItem[] = [];
+		const slottedItems = this.getSlottedNodes<MenuItem>("items");
+
+		slottedItems.forEach(item => {
+			if (item.isGroup) {
+				const groupItems = item.getSlottedNodes<MenuItem>("items");
+				items.push(...groupItems);
+			} else if (!item.isSeparator) {
+				items.push(item);
+			}
+		});
+
+		return items;
+	}
+
+	get _markSelected() {
+		return this.isSelected && this._parentItemSelectionMode !== ItemSelectionMode.None;
+	}
+
 	onBeforeRendering() {
-		const siblingsWithIcon = this._menuItems.some(menuItem => !!menuItem.icon);
+		const siblingsWithIcon = this._menuItemsAll.some(menuItem => !!menuItem.icon);
+		const itemSelectionMode = this._parentItemSelectionMode;
+
+		this._setupItemNavigation();
 
-		this._menuItems.forEach(item => {
+		this._menuItemsAll.forEach(item => {
 			item._siblingsWithIcon = siblingsWithIcon;
 		});
+
+		if (itemSelectionMode === ItemSelectionMode.None) {
+			return;
+		}
+		if (itemSelectionMode === ItemSelectionMode.SingleSelect && this.isSelected) {
+			this._parentGroup?._clearSelectedItems();
+			this.isSelected = true;
+		}
+	}
+
+	_setupItemNavigation() {
+		if (this._list) {
+			this._list._itemNavigation._getItems = () => this._menuItemsNavigation;
+		}
 	}
 
 	get _focusable() {
 		return true;
 	}
 
+	get _role() {
+		switch (this._parentItemSelectionMode) {
```

Comment:
If we need this information, let's have a private property that the group will set.

Reply:
Done